spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/ClientBuilderSslConfigurer.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.boot.ssl.jks.JksSslStoreBundle;
+import org.springframework.boot.ssl.pem.PemSslStoreBundle;
+import org.springframework.util.Assert;
+import org.springframework.util.ResourceUtils;
+
+/**
+ * Configures the SSL settings on {@link ClientBuilderSslSettings}.
+ *
+ * @author Chris Bono
+ * @since 3.2.0
+ */
+public class ClientBuilderSslConfigurer {
+
+	private final SslBundles sslBundles;
+
+	private final SslProperties sslProperties;
+
+	/**
+	 * Construct a configurer using the specified SSL bundle and properties.
+	 * @param sslBundles optional ssl bundles configured for the application
+	 * @param sslProperties optional ssl properties for the application
+	 */
+	public ClientBuilderSslConfigurer(SslBundles sslBundles, SslProperties sslProperties) {
+		this.sslBundles = sslBundles;
+		this.sslProperties = sslProperties;
+	}
+
+	/**
+	 * Applies the specified SSL properties to the specified client SSL builder.
+	 * @param clientBuilderSslSettings the ssl builder
+	 * @param clientSslProperties the ssl properties
+	 */
+	public void applySsl(ClientBuilderSslSettings<?> clientBuilderSslSettings,
+			SslConfigProperties clientSslProperties) {
+		if (!clientSslProperties.isEnabled()) {
+			return;
+		}
+		clientBuilderSslSettings.enableTls(true);
+		PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+		map.from(clientSslProperties::isVerifyHostname).to(clientBuilderSslSettings::enableTlsHostnameVerification);
+		map.from(clientSslProperties::getAllowInsecureConnection)
+			.to(clientBuilderSslSettings::allowTlsInsecureConnection);
+		if (clientSslProperties.getBundle() == null) {
+			return;
+		}
+		Assert.state(this.sslBundles != null, "SSL enabled but no SSL bundles configured");
+		Assert.state(this.sslProperties != null, "SSL enabled but no SSL properties configured");
+		String bundleName = clientSslProperties.getBundle();
+		SslBundle sslBundle = this.sslBundles.getBundle(bundleName);
+		SslOptions sslOptions = sslBundle.getOptions();
+		if (sslOptions.getCiphers() != null) {
+			Set<String> tlsCiphers = Arrays.stream(sslOptions.getCiphers()).collect(Collectors.toSet());
+			clientBuilderSslSettings.tlsCiphers(tlsCiphers);
+		}
+		if (sslOptions.getEnabledProtocols() != null) {
+			Set<String> tlsProtocols = Arrays.stream(sslOptions.getEnabledProtocols()).collect(Collectors.toSet());
+			clientBuilderSslSettings.tlsProtocols(tlsProtocols);
+		}
+		SslType sslBundleType = SslType.forSslStoreBundle(sslBundle.getStores());
+		switch (sslBundleType) {
+			case JKS -> applyJksSslProperties(clientBuilderSslSettings, map, bundleName);
+			case PEM -> applyPemSslProperties(clientBuilderSslSettings, map, bundleName);
+			case UNSUPPORTED -> throw new IllegalArgumentException(
+					"Unsupported store bundle type %s".formatted(sslBundle.getStores().getClass().getName()));
+		}
+	}
+
+	private void applyJksSslProperties(ClientBuilderSslSettings<?> clientBuilderSslSettings, PropertyMapper map,
```

Comment:
The Pulsar client builder is a bit hokey, in that it has a handful of TLS properties to set when using PEM and a handful of properties to set when using JKS. Then it takes those and under the hood handles the creation of the SSLContext. The SSLContext is not exposed to the builder.
As such I need more info that what is provided by the SSLStoreBundle (basically if I could access the Pem/JksStoreBundleDetails) I would be good). Until that time comes though, I have to dip back down into the SslProperties to get the info I need on the builder.